nf_core/modules.py

```diff
@@ -231,3 +237,481 @@ def has_valid_pipeline(self):
         if not os.path.exists(main_nf) and not os.path.exists(nf_config):
             log.error("Could not find a main.nf or nextfow.config file in: {}".format(self.pipeline_dir))
             return False
+
+
+class ModuleLint(object):
+    """
+    An object for linting modules either in a clone of the 'nf-core/modules'
+    repository or in any nf-core pipeline directory
+    """
+
+    def __init__(self, dir):
+        self.dir = dir
+        self.repo_type = self.get_repo_type()
+        self.passed = []
+        self.warned = []
+        self.failed = []
+
+    def lint(self, module=None, print_results=True, show_passed=False):
+        """
+        Lint all or one specific module
+
+        First gets a list of all local modules (in modules/local/process) and all modules
+        installed from nf-core (in modules/nf-core/software)
+
+        For all nf-core modules, the correct file structure is assured and important
+        file content is verified. If directory subject to linting is a clone of 'nf-core/modules',
+        the files necessary for testing the modules are also inspected.
+
+        For all local modules, the '.nf' file is checked for some important flags, and warnings
+        are issued if untypical content is found.
+
+        :param module:          A specific module to lint
+        :param print_results:   Whether to print the linting results
+        :param show_passed:     Whether passed tests should be shown as well
+
+        :returns:               dict of {passed, warned, failed}
+        """
+
+        # Get list of all modules in a pipeline
+        local_modules, nfcore_modules = self.get_installed_modules()
+
+        # Only lint the given module
+        # TODO --> decide whether to implement this for local modules as well
+        if module:
+            local_modules = []
+            nfcore_modules_names = [m.split("software" + os.sep)[1] for m in nfcore_modules]
+            try:
+                idx = nfcore_modules_names.index(module)
+                nfcore_modules = [nfcore_modules[idx]]
+            except ValueError as e:
+                log.error("Could not find the given module!")
+                sys.exit(1)
```

Comment:
I try to avoid doing this within the class objects, as it prevents the code from being reused anywhere else. Instead I `raise` (can use custom exceptions if you want, or just pick a type) and catch that in `__main__.py` or somewhere closer to the command line usage.

See the `nf-core sync` code for an example of this:

https://github.com/nf-core/tools/blob/8c68650409a3f95376dd55c291dae5f9fc346347/nf_core/__main__.py#L574-L578

Reply:
Great thanks for the tip, will do that!